### ISSUE TITLE
RuntimeWarning: invalid value encountered in true_divide

### ISSUE THREAD

**Issue by JaeungHyun** [openspeech-team/openspeech#157]
# ❓ Questions & Help

339.896   Total estimated model params size (MB)
Epoch 0:   1%|▎                                                 | 700/101875 [05:16<12:42:03,  2.21it/s, loss=3.19, v_num=0, train_loss=3.300, train_cross_entropy_loss=3.530, train_ctc_loss=2.750, train_wer=1.000, train_cer=1.020]
/home/bearking/voice_recognition/openspeech/openspeech/data/audio/dataset.py:175: RuntimeWarning: invalid value encountered in true_divide
  feature /= np.std(feature)
Epoch 0:   1%|▍                                                  | 825/101875 [06:10<12:37:20,  2.22it/s, loss=nan, v_num=0, train_loss=nan.0, train_cross_entropy_loss=nan.0, train_ctc_loss=nan.0, train_wer=1.000, train_cer=2.610]

warning이 발생하면 그 이후로 로스는 nan이 되어버리고 cer은 계속 증가하네요

이런 상황에서는 데이터의 어떤 부분을 체크해봐야하나요?
다른 라이브러리에서 학습할때는 이런 문제 없이 학습이 잘 되었습니다..

## Details

커맨드는 이렇습니다.

python hydra_train.py \
    dataset=ksponspeech \
    dataset.dataset_path=/home/bearking/voice_file/Training/AI_1 \
    dataset.manifest_file_path=/home/bearking/voice_recognition/KoreanSTT/transcripts.txt \
    dataset.test_dataset_path=/home/hdd/young_voice/Validation \
    dataset.test_manifest_dir=/home/hdd/young_voice/Validation \
    tokenizer=kspon_character \
    model=joint_ctc_listen_attend_spell \
    audio=melspectrogram \
    lr_scheduler=warmup_reduce_lr_on_plateau \
    trainer=gpu \
    criterion=joint_ctc_cross_entropy

**Comment by JaeungHyun** [comment]
https://github.com/openspeech-team/openspeech/blob/4ca47d698d83652e4d1f88fa142b6466b91ff400/openspeech/data/audio/dataset.py#L144-L181

해당부분에 

174번째 줄 에서 분모에 0.00000000000001과 같은 작은 수를 더해주니까 학습은 계속 진행이 되는데
로스가 음수로 가버립니다

해당부분 확인 가능할까요?

**Comment by sooftware** [comment]
오디오가 잘 로드되는게 맞나요?

**Comment by sooftware** [comment]
manifest 파일이랑 커스텀으로 만드신 것 같은데 그 정보도 같이 주시면 좋을 것 같습니다.

**Comment by JaeungHyun** [comment]
`[ 0.          0.          0.         ... -0.00146484 -0.00302124
 -0.00512695] 16000
[0.         0.         0.         ... 0.01098633 0.02313232 0.03317261] 16000
[ 0.0000000e+00 -3.0517578e-05  0.0000000e+00 ... -4.3090820e-02
 -1.9714355e-02  3.1951904e-02] 16000
[ 0.          0.          0.         ... -0.00186157  0.01171875
  0.00930786] 16000
[0.         0.         0.         ... 0.00085449 0.00082397 0.00030518] 16000
[0. 0. 0. ... 0. 0. 0.] 16000
[0. 0. 0. ... 0. 0. 0.] 16000
[ 0.0000000e+00  0.0000000e+00  0.0000000e+00 ... -3.0517578e-05
 -3.0517578e-05 -3.0517578e-05] 16000
[0.         0.         0.         ... 0.00317383 0.00366211 0.00494385] 16000
[ 0.          0.          0.         ...  0.08761597 -0.00482178
 -0.07891846] 16000
[0. 0. 0. ... 0. 0. 0.] 16000
[0.0000000e+00 0.0000000e+00 0.0000000e+00 ... 3.0517578e-05 6.1035156e-05
 9.1552734e-05] 16000
[ 0.          0.          0.         ... -0.00415039 -0.0039978
 -0.00238037] 16000
[ 0.0000000e+00  0.0000000e+00  0.0000000e+00 ...  2.4414062e-04
 -6.1035156e-05 -2.7465820e-04] 16000
[ 0.0000000e+00  0.0000000e+00  0.0000000e+00 ... -9.1552734e-05
 -9.1552734e-05 -9.1552734e-05] 16000
[ 0.0000000e+00  0.0000000e+00  0.0000000e+00 ...  0.0000000e+00
 -3.0517578e-05  0.0000000e+00] 16000
[0. 0. 0. ... 0. 0. 0.] 16000
[0. 0. 0. ... 0. 0. 0.] 16000
[ 0.          0.          0.         ...  0.00979614 -0.01980591
 -0.04641724] 16000
[0. 0. 0. ... 0. 0. 0.] 16000
[ 0.          0.          0.         ... -0.00271606 -0.0039978
 -0.00579834] 16000
[0. 0. 0. ... 0. 0. 0.] 16000
[0.         0.         0.         ... 0.00061035 0.00109863 0.00119019] 16000
[ 0.          0.          0.         ... -0.00292969  0.00064087
 -0.00039673] 16000`

load_audio에서 signal이랑 sr을 찍어보니까 나오긴합니다. 

0으로만 나오는 애들이 있네요 흐음...

**Comment by JaeungHyun** [comment]
manifest 파일은
`/home/bearking/Training/sr_16000/[원천]1.AI비서_원천_1_명령어(유소아)_training/ae_1337/ae_1337-27001-03-01-CES-M-02-A.wav	문자 바꿀래요.	80 30 4 144 514 14 16 5
/home/bearking/Training/sr_16000/[원천]1.AI비서_원천_1_명령어(유소아)_training/ae_1337/ae_1337-27002-03-01-CES-M-02-A.wav	나는 일찍 일어났어요.	10 17 4 46 210 4 46 6 172 6 16 5
/home/bearking/Training/sr_16000/[원천]1.AI비서_원천_1_명령어(유소아)_training/ae_1337/ae_1337-27003-03-01-CES-M-02-A.wav	이 동화에 공주 나와?	7 4 27 22 15 4 74 26 4 10 106 8
/home/bearking/Training/sr_16000/[원천]1.AI비서_원천_1_명령어(유소아)_training/ae_1337/ae_1337-27004-03-01-CES-M-02-A.wav	동화 목록 어디 갔지?	27 22 4 185 245 4 6 133 4 183 12 8
/home/bearking/Training/sr_16000/[원천]1.AI비서_원천_1_명령어(유소아)_training/ae_1337/ae_1337-27005-03-01-CES-M-02-A.wav	영어 뜻 알고 싶어요.	45 6 4 295 4 43 29 4 69 6 16 5
/home/bearking/Training/sr_16000/[원천]1.AI비서_원천_1_명령어(유소아)_training/ae_1337/ae_1337-27006-03-01-CES-M-02-A.wav	만화 노래 틀어 줄래?	84 22 4 32 14 4 54 6 4 78 14 8
/home/bearking/Training/sr_16000/[원천]1.AI비서_원천_1_명령어(유소아)_training/ae_1337/ae_1337-27007-03-01-CES-M-02-A.wav	집 도착이야.	91 4 61 301 7 31 5
/home/bearking/Training/sr_16000/[원천]1.AI비서_원천_1_명령어(유소아)_training/ae_1337/ae_1337-27008-03-01-CES-M-02-A.wav	공주님은 언제 나와요?	74 26 121 70 4 184 42 4 10 106 16 8
/home/bearking/Training/sr_16000/[원천]1.AI비서_원천_1_명령어(유소아)_training/ae_1337/ae_1337-27009-03-01-CES-M-02-A.wav	심심해 책 읽어 줘.	203 203 11 4 107 4 123 6 4 9 5
/home/bearking/Training/sr_16000/[원천]1.AI비서_원천_1_명령어(유소아)_training/ae_1337/ae_1337-27010-03-01-CES-M-02-A.wav	나 몇 점인가요?	10 4 148 4 146 90 13 16 8
/home/bearking/Training/sr_16000/[원천]1.AI비서_원천_1_명령어(유소아)_training/ae_1337/ae_1337-27011-03-01-CES-M-02-A.wav	갑자기 기분이 나빠지지?	237 30 18 4 18 83 7 4 10 93 12 12 8
/home/bearking/Training/sr_16000/[원천]1.AI비서_원천_1_명령어(유소아)_training/ae_1337/ae_1337-27012-03-01-CES-M-02-A.wav	사이렌이 오 분 동안 안 꺼지면 아빠한테 전화 걸어 줘.	56 7 422 7 4 28 4 83 4 27 73 4 73 4 173 12 62 4 19 93 38 67 4 51 22 4 150 6 4 9 5`

이런식으로 되어있구요


`def preprocess(dataset_path, mode='phonetic'):
    
    print('preprocess started..')

    audio_paths = list()
    transcripts = list()
    
    audio_files = glob.glob('/home/bearking/Training/sr_16000/*/*/*.wav')
    json_files = glob.glob('/home/bearking/Training/sr_16000/label/*/*.json')
    audio_files.sort()
    json_files.sort()
    
    with Parallel(n_jobs=cpu_count() - 1) as parallel:
         #do parallel and get results
        new_sentences = parallel(
            delayed(read_preprocess_text_file)(p, mode) for p in json_files)
            
        audio_paths.extend(audio_files)
        transcripts.extend(new_sentences)
    print("preprocess 성공")
    return audio_paths, transcripts`

preprocess 과정은 이렇습니다

**Comment by sooftware** [comment]
환경 문제가 의심되는데 @upskyy 테스트 통과한 디펜던시 버젼들 하나만 밑에 적어줄 수 있어요?

**Comment by upskyy** [comment]
> 0으로만 나오는 애들이 있네요 흐음...

해당 오디오에서 걸리는 것 같은데, 예외처리하고 진행해보시겠어요? @JaeungHyun

**Comment by upskyy** [comment]
테스트 통과한 디펜던시 버젼입니다. @sooftware (cc. @JaeungHyun )
```
pytorch-lightning       1.5.10
torch                   1.10.1
python-Levenshtein      0.12.2
numpy                   1.21.6
pandas                  1.1.5
astropy                 5.0.1
sentencepiece           0.1.96
hydra-core              1.0.7
wget                    3.2
wandb                   0.12.10
```

**Comment by sooftware** [comment]
아 일부 오디오가 문제 있는 경우 저럴 수 있어요

**Comment by JaeungHyun** [comment]
@upskyy 파일이 특정되지 않는데 걸러낼 방법이 있을까요

**Comment by upskyy** [comment]
@JaeungHyun 다양한 방법이 있겠지만, 가장 쉬운 방법은 코드 level에서 모두 0인 부분 삭제하는게 떠오르네요.

**Comment by JaeungHyun** [comment]
@upskyy 그러면 dimension  문제가 일어나지 않을까 싶어서요

**Comment by upskyy** [comment]
@JaeungHyun 학습 들어가기전에 전처리 과정으로 manifest 파일에서 삭제하고 진행하면 되지 않을까요? 제가 해당 파일이 없어서 재현이 어렵네요.

**Comment by JaeungHyun** [comment]
해당 경고가 뜨는 파일을 제거하니 문제가 해결되었습니다
